### PR TITLE
fix(monorepo): upgrade pnpm-lock.yaml to 9.0

### PR DESCRIPTION
**Pull request by jakim929**
see major changes in https://github.com/pnpm/pnpm/releases/tag/v9.0.0

https://github.com/ethereum-optimism/ecosystem/pull/274 this PR upgraded our package manager to pnpm@9 but **did not** update the lockfile version to v9 (it was previously v6). was causing a lot of brittleness in CI / docker builds.

### 1. fixes issue where dependency "install" scripts require a dependent bin
this required deleting `node_modules` and `pnpm-lock` and regenerating the `pnpm-lock.yaml` file. Otherwise, the `hasBin` field inside pnpm-lock doesn't get correctly updated for packages like [this](https://github.com/websockets/bufferutil/blob/853a377b618c3daf62e5be490cbb5286b18f358e/package.json#L10) where the install script requires a dependency, causing docker build failures like `node-gyp-build: not found`

### 2. fix type issues with trpc
not exactly sure, but peer-dependency resolution logic might have changed? required adding some peer dependency packages explicitly (which is weird because `auto-install-peers` should be `true`). can investigate later

### 3. update internal packages with workspace: prefix
with the new pnpm v9, `link-workspace-packages` is `false` by default, meaning we need to add `workspace:` to the package version if we want to use an internal package. It's better anyway to make local packages more explicit now, since we may start publishing packages to npm.

### CODE REVIEW

**Comment by jakim929** [comment]
* **#277** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/277?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jakim929 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/op-app/package.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
     "tsup": "^8.0.1"
   },
   "devDependencies": {
-    "@tanstack/react-query": "^5.10.0",
+    "@eth-optimism/superchain-registry": "github:ethereum-optimism/superchain-registry#0496fcd6ee4e36f2bedd3e4fbffa9146b10219af",
```
Comment (jakim929) [inline packages/op-app/package.json:28]: this works again 🔥 seems like it was an issue with the lockfile

File: apps/dapp-console/package.json [hunk 2]
```diff
@@ -13,14 +13,17 @@
...
     "@growthbook/growthbook-react": "^0.24.0",
     "@hookform/resolvers": "^3.3.4",
     "@privy-io/react-auth": "^1.59.5",
     "@remixicon/react": "^4.1.1",
     "@sentry/nextjs": "^7.106.0",
+    "@tanstack/react-query": "^4",
     "@trpc/client": "^10.45.2",
     "@trpc/next": "^10.45.2",
+    "@trpc/react-query": "^10.45.2",
+    "@trpc/server": "^10.45.2",
```
[the remark this answers is not in the document]
Comment (nitaliano) [reply apps/dapp-console/package.json:26]: I guess there's no way around it with them being marked as peer deps.

**Review by nitaliano — APPROVED** [review]

**Comment by jeffzwang** [comment]
🔥

**Pull request merged**